frontend/app/helpers/login.helper.ts

```diff
@@ -0,0 +1,24 @@
+import { type UserCredentials, type ValidationError } from "@/app/login/page"
+
+export const LoginValidation = (
+  credentials: UserCredentials,
+  setValidationError: React.Dispatch<React.SetStateAction<ValidationError>>
+): void => {
+  const regexPatternUsername = /^[a-zA-Z0-9_]{3,20}$/
+  const regexPatternPassword =
+    /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/
+  const isUsernameValid = regexPatternUsername.test(credentials.username)
+  const isPasswordValid = regexPatternPassword.test(credentials.password)
+
+  setValidationError(
+    (prevError: ValidationError): ValidationError => ({
+      ...prevError,
```

Comment:
Why do we need this?

Reply:
sorry we do not need it because if everytime check for both values. so no previous data required